### PR TITLE
Docs: Describe jQuery.when split paths

### PR DESCRIPTION
**Pull request by gibson042**
Ref https://github.com/jquery/jquery/commit/356a3bccb0e7468a2c8ce7d8c9c6cd0c5d436b8b

### CODE REVIEW

File: pages/upgrade-guide/3.0.md [hunk 1]
```diff
@@ -347,14 +347,24 @@ https://github.com/jquery/jquery/issues/1722
...
 
-As of jQuery 3.0, the `$.when()` method no longer passes along progress notifications from input Deferreds to the output Deferred. Progress messages are not part of the Promises/A+ specification, and the behavior of progress notifications in `$.when()` were not previously documented in the jQuery API.
+Further, a more clear distinction is now recognized between multi-argument calls to `$.when` and single- or no-argument calls.
+Multi-argument calls behave similarly to `Promise.all`, aggregating fulfillment values into a fulfillment array (with jQuery-specific enhancements of also aggregating fulfillment contexts and supporting multi-valued fulfillments), or alternatively rejecting with the first rejection value.
+Single- and no-argument calls behave similarly to `Promise.resolve`, returning a Deferred that resolves identically to thenable or Promise-like input, or fulfills with its non-Promise input (as appropriate).
+As of jQuery 3.0, both of these return a new Deferred (previous versions did not create a new Deferred when called with a single Deferred input).
 
-https://github.com/jquery/jquery/issues/2710  
+https://github.com/jquery/jquery/issues/2018
```
Comment (dmethvin) [inline pages/upgrade-guide/3.0.md]: Do these have two spaces after them to create a line break? I can't tell.

File: pages/upgrade-guide/3.0.md [hunk 2]
```diff
@@ -347,14 +347,24 @@ https://github.com/jquery/jquery/issues/1722
...
+Multi-argument calls behave similarly to `Promise.all`, aggregating fulfillment values into a fulfillment array (with jQuery-specific enhancements of also aggregating fulfillment contexts and supporting multi-valued fulfillments), or alternatively rejecting with the first rejection value.
+Single- and no-argument calls behave similarly to `Promise.resolve`, returning a Deferred that resolves identically to thenable or Promise-like input, or fulfills with its non-Promise input (as appropriate).
+As of jQuery 3.0, both of these return a new Deferred (previous versions did not create a new Deferred when called with a single Deferred input).
 
-https://github.com/jquery/jquery/issues/2710  
+https://github.com/jquery/jquery/issues/2018  
 https://github.com/jquery/jquery/issues/2546  
+https://github.com/jquery/jquery/issues/3029  
+
+#### Breaking change: $.when() progress notifications
```
Comment (dmethvin) [inline pages/upgrade-guide/3.0.md]: :+1: Good call on breaking this out

File: pages/upgrade-guide/3.0.md [hunk 3]
```diff
@@ -347,14 +347,24 @@ https://github.com/jquery/jquery/issues/1722
...
 
-As of jQuery 3.0, the `$.when()` method no longer passes along progress notifications from input Deferreds to the output Deferred. Progress messages are not part of the Promises/A+ specification, and the behavior of progress notifications in `$.when()` were not previously documented in the jQuery API.
+Further, a more clear distinction is now recognized between multi-argument calls to `$.when` and single- or no-argument calls.
+Multi-argument calls behave similarly to `Promise.all`, aggregating fulfillment values into a fulfillment array (with jQuery-specific enhancements of also aggregating fulfillment contexts and supporting multi-valued fulfillments), or alternatively rejecting with the first rejection value.
+Single- and no-argument calls behave similarly to `Promise.resolve`, returning a Deferred that resolves identically to thenable or Promise-like input, or fulfills with its non-Promise input (as appropriate).
+As of jQuery 3.0, both of these return a new Deferred (previous versions did not create a new Deferred when called with a single Deferred input).
```
Comment (dmethvin) [inline pages/upgrade-guide/3.0.md]: Should we make any assertions about the result being sync or async?

Comment (gibson042) [inline pages/upgrade-guide/3.0.md]: I think any relevant changes are covered in the Deferred section, especially if https://github.com/jquery/jquery/pull/3105 lands.

File: pages/upgrade-guide/3.0.md (hunk 1 above)
Comment (gibson042) [inline pages/upgrade-guide/3.0.md]: Yes. I didn't realize why they were present until their removal borked the GitHub rich diff.

Comment (dmethvin) [inline pages/upgrade-guide/3.0.md]: Yeah and they call html markup obscure. :smiling_imp: Another option would be to put them all on one line but turn them into links, something like `Related: gh-1234, gh-2345`. That could all be reformatted in a later pass though.

**Pull request closed without merging**